src/Type.h

```diff
@@ -84,10 +84,11 @@ struct halide_handle_cplusplus_type {
     /// One set of modifiers on a type.
     /// The const/volatile/restrict properties are "inside" the pointer property.
     enum Modifier : uint8_t {
-        Const = 1 << 0,     ///< Bitmask flag for "const"
-        Volatile = 1 << 1,  ///< Bitmask flag for "volatile"
-        Restrict = 1 << 2,  ///< Bitmask flag for "restrict"
-        Pointer = 1 << 3,   ///< Bitmask flag for a pointer "*"
+        Const = 1 << 0,            ///< Bitmask flag for "const"
+        Volatile = 1 << 1,         ///< Bitmask flag for "volatile"
+        Restrict = 1 << 2,         ///< Bitmask flag for "restrict"
+        Pointer = 1 << 3,          ///< Bitmask flag for a pointer "*"
+        FunctionPointer = 1 << 4,  ///< Bitmask flag for a function pointer (if set, Pointer will also be set)
```

Comment:
If Pointer is also going to be set, I'm not sure that word should be repeated in the name here. On the one hand, this seems fine, on the other, I'm not sure if the designation for a function pointer is really right as we don't have anything like all the information about the type. So far as I can tell, we're using this to avoid putting the pointer annotation (*) in because the function pointer is actually a typedef that already contains the indirection, as they always do, and the type printing just happens to work in this way. To this end, I think the right name for this might be ```FunctionTypedef``` and the comment can indicate it should always have the ```Pointer``` flag set as well.

Reply:
Done